storage/json/datasource.go

```diff
@@ -80,11 +80,24 @@ func (ds *DataSource) Get(ctx context.Context, variables octosql.Variables, stre
 		batchSize:                     ds.batchSize,
 	}
 
+	ctx, cancel := context.WithCancel(ctx)
+	rs.workerCtxCancel = cancel
+	rs.workerCloseErrChan = make(chan error)
```

Comment:
It's better to make this channel buffered with capacity of 1. This way if the worker errors and there's nobody to read the channel, the goroutine won't hang on channel send.